holidays/countries/azerbaijan.py

```diff
@@ -115,19 +115,19 @@ def _add_observed(dt: date, name: str = None):
 
         if year >= 1993:
             name = "Ramazan Bayrami"
-            observed_dates.update(self._add_eid_al_fitr_day(name))
-            observed_dates.update(self._add_eid_al_fitr_day_two(name))
+            dts_observed.update(self._add_eid_al_fitr_day(name))
+            dts_observed.update(self._add_eid_al_fitr_day_two(name))
 
             name = "Gurban Bayrami"
-            observed_dates.update(self._add_eid_al_adha_day(name))
-            observed_dates.update(self._add_eid_al_adha_day_two(name))
+            dts_observed.update(self._add_eid_al_adha_day(name))
+            dts_observed.update(self._add_eid_al_adha_day_two(name))
 
         # Article 105 of the Labor Code of the Republic of Azerbaijan states:
         # 5. If interweekly rest days and holidays that are not considered
         # working days overlap, that rest day is immediately transferred to
         # the next working day.
         if self.observed and year >= 2006:
-            hol_dates = observed_dates.union(non_observed_dates)
+            dts_observed = dts_observed.union(dts_non_observed)
```

Comment:
A separate set (`hol_dates`) is needed here, as `dts_observed + dts_non_observed`, and `dts_observed` should remain unchanged. `_add_observed()` also should check for `... in in hol_dates` and add to `hol_dates`.

Reply:
This has been addressed too.